app/controllers/articles_controller.rb

```diff
@@ -127,16 +127,8 @@ def update
     authorize @article
     @user = @article.user || current_user
 
-    not_found if @article.user_id != @user.id && !@user.has_role?(:super_admin)
```

Comment:
Other articles should be filtered out by `authorize`, so no need to check one more time.

Reply:
Not part of this PR, but this got me wondering if in the long run, we should actually use Pundit's `policy_scope` and get rid of the `before_action` that sets `@article`.